profile_test.go

```diff
@@ -0,0 +1,313 @@
+	assert.Nil(t, err)
+	assert.Equal(t, expectedOID, inputOID)
+
+	// Section 3.1 of https://tools.ietf.org/html/draft-ietf-cbor-tags-oid-05#section-3
+	inputOID = "2.16.840.1.101.3.4.2.1"
+
+	// CBOR Input
+	input = []byte{
+		0x49, 0x60, 0x86, 0x48, 0x01, 0x65, 0x03, 0x04, 0x02, 0x01,
+	}
+	err = profile.UnmarshalCBOR(input)
+	assert.Nil(t, err)
+	expectedOID, err = profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedOID, inputOID)
+}
+
+// TestProfile_MarshalJSON_URL tests the JSON Marshaling for a known URL
+func TestProfile_MarshalJSON_URL(t *testing.T) {
+	inputURL := "http://example.com"
+	profile, err := NewProfile(inputURL)
+	assert.Nil(t, err)
+
+	expected := []byte{
+		0x22, 0x68, 0x74, 0x74, 0x70, 0x3a, 0x2f, 0x2f, 0x65, 0x78,
+		0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0x22,
+	}
+
+	actual, err := profile.MarshalJSON()
+	assert.Nil(t, err)
+	assert.Equal(t, expected, actual)
+}
+
+// TestProfile_UnMarshalJSON_URL tests the UnMarshaling of a JSON value as URL string
+func TestProfile_UnMarshalJSON_URL(t *testing.T) {
+	inputURL := "http://example.com"
+	profile, err := NewProfile(inputURL)
+	assert.Nil(t, err)
+
+	data := []byte{
+		0x22, 0x68, 0x74, 0x74, 0x70, 0x3a, 0x2f, 0x2f, 0x65, 0x78,
+		0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0x22,
+	}
+
+	err = profile.UnmarshalJSON(data)
+	assert.Nil(t, err)
+	recvURL, err := profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, inputURL, recvURL)
+
+	// Negative Test Cases
+	// Corrupted Header
+	data = []byte{
+		0x10, 0x68, 0x74, 0x74, 0x70, 0x3a, 0x2f, 0x2f, 0x65, 0x10,
+	}
+	err = profile.UnmarshalJSON(data)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "invalid character")
+
+	// Invalid Input data as string
+	data = []byte{
+		0x22, 0x88, 0x78, 0x74, 0x70, 0x3a, 0x2f, 0x2f, 0x65, 0x78,
+		0x61, 0x6d, 0x70, 0x6c, 0x65, 0x2e, 0x63, 0x6f, 0x6d, 0x22,
+	}
+	err = profile.UnmarshalJSON(data)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "failed to extract OID")
+
+}
+
+// TestProfile_MarshalJSON_OID validates the JSON Marshaling of OID as profile
+func TestProfile_MarshalJSON_OID(t *testing.T) {
+	inputOID := "2.5.2.1"
+	profile, err := NewProfile(inputOID)
+	assert.Nil(t, err)
+	expected := []byte(`"2.5.2.1"`)
+	actual, err := profile.MarshalJSON()
+	assert.Nil(t, err)
+	assert.Equal(t, expected, actual)
+}
+
+// TestProfile_UnMarshalJSON_OID tests the JSON Unmarshaling for OID as profile
+func TestProfile_UnMarshalJSON_OID(t *testing.T) {
+	inputOID := "1.2.3.4"
+	profile, err := NewProfile(inputOID)
+	assert.Nil(t, err)
+	input := []byte(`"2.5.2.1"`)
+	err = profile.UnmarshalJSON(input)
+	assert.Nil(t, err)
+	expectedOID := "2.5.2.1"
+	receivedOID, err := profile.Get()
+	assert.Nil(t, err)
+	assert.Equal(t, expectedOID, receivedOID)
+
+	// Partial OID test
+	inputOID = ".2.3.4"
+	_, err = NewProfile(inputOID)
+	assert.NotNil(t, err)
+	assert.Contains(t, err.Error(), "failed to extract OID from string")
```

Comment:
these two lines could be made into a single `assert.EqualError` call

Reply:
Done!